cs-rin-ru-enhanced-mod.user.js

```diff
@@ -59,30 +62,65 @@ const FORUM_BASE_URL = getBaseUrl();
 //Contains the list of friends
 const FRIENDS_LIST = [];
 
-const CONNECTED = document.querySelector("#menubar > table:nth-child(3) > tbody > tr > td:nth-child(2) > a:nth-child(2)").text !== ' Login';
+const CONNECTED = document.querySelector("#menubar > table:nth-child(3) > tbody > tr > td:nth-child(2)").lastElementChild.getAttribute("href").match("mode=login") == null
 
 const USERNAME = $("#menubar > table:nth-child(3) > tbody > tr > td:nth-child(2) > a:nth-child(2)")[0].textContent.slice(10, -2);
 
+// Declare a promise to wait for the variable to be updated
+let updatePromise = null;
 //Retrieve friends list
 async function retrievesFriendsLists() {
-    await fetch(FORUM_BASE_URL + "ucp.php?i=zebra&mode=friends")
-        .then(response => response.text())
-        .then(text => {
-            let parser = new DOMParser();
-            let doc = parser.parseFromString(text, "text/html");
-            FRIENDS_LIST.push(...Array.from(doc.querySelector('#ucp > table > tbody > tr:nth-child(3) > td.row2 > select').children, node => node.innerText));
+    // Checks if the promise is already being executed
+    if (!updatePromise) {
+        // Create a new promise
+        updatePromise = new Promise(async (resolve, reject) => {
+            // Check if the friends list has already been updated
+            if (FRIENDS_LIST.length === 0) {
+                try {
+                    // Performs the query to retrieve the list of friends
+                    const response = await fetch(FORUM_BASE_URL + "ucp.php?i=zebra&mode=friends");
+                    const text = await response.text();
+                    // Parse the answer to extract the list of friends
+                    let parser = new DOMParser();
+                    let doc = parser.parseFromString(text, "text/html");
+                    FRIENDS_LIST.push(...Array.from(doc.querySelector('#ucp > table > tbody > tr:nth-child(3) > td.row2 > select').children, node => node.innerText));
```

Comment:
This throws an error if you have no friends added.
The HTML looks like this:
`<td class="row2" align="center"><b class="genmed">No friends currently defined</b></td>`

Reply:
I'm not sure, I'm not using this feature but maybe we can just check if 'select' element exists and if not return empty FRIENDS_LIST?